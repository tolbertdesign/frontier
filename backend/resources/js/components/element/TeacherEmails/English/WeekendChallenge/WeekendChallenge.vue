<template>
  <p>
    To: School Families
    <br>When: Day Before Weekend Begins (typically Friday)
    <br>Subject: Special Weekend Challenge!
    <br><br>{{ props.event_name }} Families,
    <br><br>Thank you again for supporting our school through the <strong>{{ props.event_name }}!</strong>
    <br><br>This weekend we are issuing a special challenge! Your child can earn a bonus prize simply by getting a qualifying pledge
    <strong>this weekend only.</strong> Check out <a :href="`https://${props.link}`">{{ props.link }}</a> for details!
    <br><br>And if you're looking for the easiest way to ask for a pledge, just upload a photo
    of your student to <a :href="`https://${props.link}`">{{ props.link }}</a>
    to create a personalized Student Star Video. This fun, hilarious video starring your child can be shared by
    email or social media to connect with sponsors and ask for a pledge.
    <br><br>Thank you again for helping our class!
    <br><br>P.S. - If you haven't registered your child on <a :href="`https://${props.link}`">{{ props.link }}</a>,
    search for your school or use our school registration code: <strong>{{ props.registration_code }}</strong>
    <br>
  </p>
</template>

<script>
export default {
  name: 'WeekendChallenge',
  status: 'prototype',
  release: '1.0.0',
  props: {
    props: {
      type: Object,
      default: () => {},
    },
  },
}
</script>
