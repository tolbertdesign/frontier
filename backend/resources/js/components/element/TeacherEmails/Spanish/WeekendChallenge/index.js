import WeekendChallenge from './WeekendChallenge.vue'

export default WeekendChallenge
