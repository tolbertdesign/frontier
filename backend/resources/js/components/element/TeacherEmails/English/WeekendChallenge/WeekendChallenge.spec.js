import { shallowMount, createLocalVue } from '@vue/test-utils'
import WeekendChallenge from './WeekendChallenge.vue'

const localVue = createLocalVue()

const props = {
  event_name: 'event name',
  registration_code: 'registration code',
  due_date: 'due date',
  link: 'example.com',
  class_pledge_total: '500',
}

describe('WeekendChallenge', () => {
  it('Snapshop test for WeekendChallenge', () => {
    const wrapper = shallowMount(WeekendChallenge, {
      localVue,
      propsData: {
        props: props,
      },
    })
    expect(wrapper).toMatchSnapshot()
  })
})
