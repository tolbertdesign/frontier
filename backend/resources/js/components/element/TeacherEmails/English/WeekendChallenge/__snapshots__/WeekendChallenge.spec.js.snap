// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeekendChallenge Snapshop test for WeekendChallenge 1`] = `
<p>
  To: School Families
  <br>When: Day Before Weekend Begins (typically Friday)
  <br>Subject: Special Weekend Challenge!
  <br><br>event name Families,
  <br><br>Thank you again for supporting our school through the <strong>event name!</strong> <br><br>This weekend we are issuing a special challenge! Your child can earn a bonus prize simply by getting a qualifying pledge
  <strong>this weekend only.</strong> Check out <a href="https://example.com">example.com</a> for details!
  <br><br>And if you're looking for the easiest way to ask for a pledge, just upload a photo
  of your student to <a href="https://example.com">example.com</a>
  to create a personalized Student Star Video. This fun, hilarious video starring your child can be shared by
  email or social media to connect with sponsors and ask for a pledge.
  <br><br>Thank you again for helping our class!
  <br><br>P.S. - If you haven't registered your child on <a href="https://example.com">example.com</a>,
  search for your school or use our school registration code: <strong>registration code</strong> <br></p>
`;
